feat(protocol): Add input validation for voltage and current setpoints

avion23 opened the pull request:
## Input Validation for Safety

Adds validation to prevent out-of-range values from being written to hardware DACs.

### Changes
- `protocol_handler.c`: Validate parsed command values before processing
- `pwrctl.c`: Add bounds checking (65V max input, 60V max output, 20A max current)  
- `opendps.c`: Validate calibration coefficients (current ADC gain: 0.1-100.0f)

### Behavior
- Invalid commands return error response
- Hardware limits prevent unsafe DAC writes
- Device maintains last valid state on invalid input

Prevents potential hardware damage from malformed commands or invalid calibration data.

kanflo commented on opendps/protocol_handler.c:
```diff
@@ -488,16 +482,9 @@ static command_status_t handle_upgrade_start(frame_t *frame)
 static command_status_t handle_change_screen(frame_t *frame)
 {
     emu_printf("%s\n", __FUNCTION__);
-    uint8_t cmd, screen_id;
```
While my way of dealing with the protocol might be considered slight overkill, this should be a separate PR I think.

kanflo commented:
While I am all for input validation, I would like to have all protocol handling changes in a separate PR.

Commit:
feat(safety): Add bounds checking for voltage, current and calibration values

Adds validation to prevent out-of-range values from being written to hardware
or stored as calibration coefficients.

Changes:
- pwrctl.c: Add sanity limits (65V max input, 60V max output, 20A max current)
- opendps.c: Validate calibration coefficients with reasonable bounds

Invalid values are rejected, maintaining the last valid state.
Prevents potential hardware damage from invalid ADC readings or calibration data.

avion23 commented:
Thanks for merging the first one. 
I reduced this to only some validations, completely removed protocol changes.
